ArcGISRuntimeSDKQt_CppSamples/EditData/SnapGeometryEdits/SnapGeometryEdits.h

```diff
@@ -0,0 +1,118 @@
+// [WriteFile Name=SnapGeometryEdits, Category=EditData]
+// [Legal]
+// Copyright 2024 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef SNAPGEOMETRYEDITS_H
+#define SNAPGEOMETRYEDITS_H
+
+namespace Esri::ArcGISRuntime {
+  class GeometryEditor;
+  class GeometryEditor;
+  class Graphic;
+  class GraphicsOverlay;
+  class Map;
+  class MapQuickView;
+  class Portal;
+  class PortalItem;
+  class SimpleFillSymbol;
+  class SimpleLineSymbol;
+  class SimpleMarkerSymbol;
+  class SnapSourceSettings;
+}
+
+#include <QObject>
+
+Q_MOC_INCLUDE("MapQuickView.h");
+
+class SnapGeometryEdits : public QObject
+{
+  Q_OBJECT
+
+  Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(bool geometryEditorStarted READ geometryEditorStarted NOTIFY geometryEditorStartedChanged)
+  Q_PROPERTY(bool canUndo READ canUndo NOTIFY canUndoChanged)
+  Q_PROPERTY(bool elementIsSelected READ elementIsSelected NOTIFY elementIsSelectedChanged)
+  Q_PROPERTY(bool layersLoaded MEMBER m_layersLoaded NOTIFY layersLoadedChanged)
+  Q_PROPERTY(QList<QString> pointLayers MEMBER m_pointLayers NOTIFY pointLayersChanged)
+  Q_PROPERTY(QList<QString> polylineLayers MEMBER m_polylineLayers NOTIFY polylineLayersChanged)
```

Comment:
```suggestion
  Q_PROPERTY(QStringList pointLayers MEMBER m_pointLayers NOTIFY pointLayersChanged)
  Q_PROPERTY(QStringList polylineLayers MEMBER m_polylineLayers NOTIFY polylineLayersChanged)
```

And other associated changes that should be made as well.